pkg/ingester/ingester_test.go

```diff
@@ -503,190 +505,245 @@ func TestIngesterAppendBlankLabel(t *testing.T) {
-		require.NoError(t, os.RemoveAll(dirname))
+		require.NoError(t, os.RemoveAll(dir))
 	}()
 
-	cfg := defaultIngesterTestConfig()
-	cfg.WALConfig.WALEnabled = true
-	cfg.WALConfig.Recover = true
-	cfg.WALConfig.Dir = dirname
-	cfg.WALConfig.CheckpointDuration = 100 * time.Minute
-
-	limits := defaultLimitsTestConfig()
-	limits.MaxLocalSeriesPerUser = 1
-	limits.MaxLocalMetricsWithMetadataPerUser = 1
+	chunksDir := filepath.Join(dir, "chunks")
+	blocksDir := filepath.Join(dir, "blocks")
+	require.NoError(t, os.Mkdir(chunksDir, os.ModePerm))
+	require.NoError(t, os.Mkdir(blocksDir, os.ModePerm))
 
-	_, ing := newTestStore(t, cfg, defaultClientTestConfig(), limits, nil)
+	chunksIngesterGenerator := func() *Ingester {
+		cfg := defaultIngesterTestConfig()
+		cfg.WALConfig.WALEnabled = true
+		cfg.WALConfig.Recover = true
+		cfg.WALConfig.Dir = chunksDir
+		cfg.WALConfig.CheckpointDuration = 100 * time.Minute
 
-	userID := "1"
-	// Series
-	labels1 := labels.Labels{{Name: labels.MetricName, Value: "testmetric"}, {Name: "foo", Value: "bar"}}
-	sample1 := client.Sample{
-		TimestampMs: 0,
-		Value:       1,
-	}
-	sample2 := client.Sample{
-		TimestampMs: 1,
-		Value:       2,
-	}
-	labels3 := labels.Labels{{Name: labels.MetricName, Value: "testmetric"}, {Name: "foo", Value: "biz"}}
-	sample3 := client.Sample{
-		TimestampMs: 1,
-		Value:       3,
-	}
-	// Metadata
-	metadata1 := &client.MetricMetadata{MetricName: "testmetric", Help: "a help for testmetric", Type: client.COUNTER}
-	metadata2 := &client.MetricMetadata{MetricName: "testmetric2", Help: "a help for testmetric2", Type: client.COUNTER}
-
-	// Append only one series and one metadata first, expect no error.
-	ctx := user.InjectOrgID(context.Background(), userID)
-	_, err = ing.Push(ctx, client.ToWriteRequest([]labels.Labels{labels1}, []client.Sample{sample1}, []*client.MetricMetadata{metadata1}, client.API))
-	require.NoError(t, err)
+		_, ing := newTestStore(t, cfg, defaultClientTestConfig(), limits, nil)
+		return ing
+	}
 
-	testLimits := func() {
-		// Append to two series, expect series-exceeded error.
-		_, err = ing.Push(ctx, client.ToWriteRequest([]labels.Labels{labels1, labels3}, []client.Sample{sample2, sample3}, nil, client.API))
-		if resp, ok := httpgrpc.HTTPResponseFromError(err); !ok || resp.Code != http.StatusTooManyRequests {
-			t.Fatalf("expected error about exceeding metrics per user, got %v", err)
-		}
-		// Append two metadata, expect no error since metadata is a best effort approach.
-		_, err = ing.Push(ctx, client.ToWriteRequest(nil, nil, []*client.MetricMetadata{metadata1, metadata2}, client.API))
+	blocksIngesterGenerator := func() *Ingester {
+		ing, err := newIngesterMockWithTSDBStorageAndLimits(defaultIngesterTestConfig(), limits, blocksDir, nil)
 		require.NoError(t, err)
+		require.NoError(t, services.StartAndAwaitRunning(context.Background(), ing))
+		// Wait until it's ACTIVE
+		test.Poll(t, 10*time.Millisecond, ring.ACTIVE, func() interface{} {
```

Comment:
Let's be more generous to avoid flaky tests.

```suggestion
		test.Poll(t, time.Second, ring.ACTIVE, func() interface{} {
```